govcd/openapi_test.go

```diff
@@ -0,0 +1,261 @@
+// +build functional openapi ALL
+
+package govcd
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+
+	. "gopkg.in/check.v1"
+)
+
+// Test_OpenAPIRawJsonAudiTrail uses low level GET function to test out that pagination really works. It is an example
+// how to fetch response from multiple pages in RAW json messages without having defined as struct.
+func (vcd *TestVCD) Test_OpenAPIRawJsonAudiTrail(check *C) {
+	minimumRequiredApiVersion := "33.0"
+	skipOpenApiEndpointTest(vcd, check, "1.0.0/auditTrail", minimumRequiredApiVersion)
+
+	urlRef, err := vcd.client.Client.OpenApiBuildEndpoint("1.0.0/auditTrail")
+	check.Assert(err, IsNil)
+
+	// Limit search of audits trails to the last 12 hours so that it doesn't take too long and set pageSize to be 1 result
+	// to force following pages
+	queryParams := url.Values{}
+	filterTime := time.Now().Add(-12 * time.Hour).Format(types.FiqlQueryTimestampFormat)
+	queryParams.Add("filter", "timestamp=gt="+filterTime)
+	queryParams.Add("pageSize", "1")
+
+	allResponses := []json.RawMessage{{}}
+	err = vcd.vdc.client.OpenApiGetAllItems(minimumRequiredApiVersion, urlRef, queryParams, &allResponses)
+
+	check.Assert(err, IsNil)
+	check.Assert(len(allResponses) > 1, Equals, true)
+
+	// Build a regex ant match it internally so that we are sure auditTrail events are returned in RAW json message. There
+	// should be the same amount of audit event IDs as total responses
+	auditLogUrn := regexp.MustCompile("urn:vcloud:audit:")
+	responseStrings := jsonRawMessagesToStrings(allResponses)
+	allStringResponses := `[` + strings.Join(responseStrings, ",") + `]`
+	matches := auditLogUrn.FindAllStringIndex(allStringResponses, -1)
+	check.Assert(len(matches), Equals, len(allResponses))
+}
+
+// Test_OpenAPIInlineStructAudiTrail uses low level GET function to test out that get function can unmarshal directly
+// to user defined inline type
+func (vcd *TestVCD) Test_OpenAPIInlineStructAudiTrail(check *C) {
```

Comment:
Nit. In some places capital letters `OpenAPI` is used. Instead, we should use camel-case `OpenApi` consistently.

Reply:
True. Partly because many of linters suggest to use OpenAPI (abbreviations to be in capital), but I have done search/replace so there should be no outstanding cases

Reply:
I have changed to use OpenAPI casing in docs and OpenApi in function names.